### PR TITLE
Add peer_set class

### PR DESCRIPTION
**Pull request by miguel1117**
- Add a peer_set class to represent a collection of peers
- Add incoming attribute to the Peer

### CODE REVIEW

File: peer_set.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def find(self, element: Peer) -> Optional[Peer]:
+        for p in self:
+            if element == p:
+                return p
+
+        return None
+
+    def cleanup_inactive(self, inactivity_threshold_seconds: int, verbosity: int):
+        for peer in self.copy():
+            if peer.last_seen is None:
```
Comment (dsiganos) [inline peer_set.py:32]: But what if a peer is never seen? It will never be removed?

Comment (dsiganos) [reply peer_set.py:32]: Should we consider peer construction as been seen?

File: peer_set.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from time import time
+from typing import Iterable, Optional
+
+from pynanocoin import Peer
+
+
+class peer_set(set):
```
Comment (dsiganos) [inline peer_set.py:7]: What about thread protection. Don;t we need a mutex?

Comment (dsiganos) [reply peer_set.py:7]: It seems that peer_set is not thread protected and expects the caller to do thread protection.

File: peer_set.py (hunk 1 above)
Comment (miguel1117) [reply peer_set.py:32]: > Should we consider peer construction as been seen?

Yeah, that's probably the easiest way.

File: pynanocoin.py [hunk 3]
```diff
@@ -276,15 +276,15 @@ def __str__(self):
...
+    def __init__(self, ip=ip_addr(), port=0, score=-1, is_voting=False, last_seen: int = None, incoming=False):
         assert isinstance(ip, ip_addr)
         self.ip = ip
         self.port = port
         self.peer_id = None
         self.is_voting = is_voting
         self.telemetry = None
         self.aux = {}
-        self.last_seen = last_seen
+        self.last_seen = last_seen if last_seen else int(time.time())
```
[the remark this answers is not in the document]
Comment (miguel1117) [reply pynanocoin.py:287]: Oh right, I forgot that default values in Python don't have to be constants.

**Review by dsiganos — APPROVED** [review]

**Pull request merged**